packages/react-instantsearch-hooks/src/hooks/__tests__/useConnector.test.tsx

```diff
@@ -659,4 +663,66 @@ describe('useConnector', () => {
     expect(indexContext.current!.removeWidgets).toHaveBeenCalledTimes(1);
     expect(indexContext.current!.addWidgets).toHaveBeenCalledTimes(2);
   });
+
+  describe('parentIndexId', () => {
+    test('uses the parentIndexId when set', async () => {
+      const searchClient = createSearchClient({});
+      const { InstantSearchSpy, indexContext } = createInstantSearchSpy();
+
+      function App() {
+        return (
+          <StrictMode>
+            <InstantSearchSpy searchClient={searchClient} indexName="root">
+              <Index indexName="1">
+                <Index indexName="2" parentIndexId={null}>
+                  <CustomWidget attribute="2-natural" />
+                  <CustomWidget parentIndexId="1" attribute="1-moved" />
+                </Index>
+                <CustomWidget attribute="1-natural" />
+                {/* @TODO: parentIndexId 2 isn't found? */}
+                {/* <CustomWidget parentIndexId="2" attribute="2-moved" /> */}
```

Comment:
we get an error that index 2 can't be found. It seems like CustomWidget gets rendered when only its parents are created, not yet the other components. Maybe that's specific StrictMode behaviour, but it would be discounting the feature of parentIndexId with a different value than `null` quite a bit.